plugin/forward/forward.go

```diff
@@ -177,6 +177,9 @@ func (f *Forward) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg
 	}
 
 	if upstreamErr != nil {
+		serverFailureErr := new(dns.Msg)
```

Comment:
this is done here: https://github.com/coredns/coredns/blob/master/core/dnsserver/server.go#L250

Reply:
In that case it is done to late in the flow for other plugin to use.
Namely metrics and apm.
It make sense to do it at the end of the chain as a sort of catch all.but the next plugin apparently do not have the info. unless i go to option 2 by changing the usage of the [ metrics/handler.go:ServeDns](https://github.com/coredns/coredns/blob/master/plugin/metrics/handler.go#L29)
And possibly in the [trace/trace.go:ServeDns](https://github.com/coredns/coredns/blob/master/plugin/trace/trace.go#L126)
Or am i missing something.